sentinel-dashboard/src/main/java/com/alibaba/csp/sentinel/dashboard/controller/v2/ParamFlowRuleControllerV2.java

```diff
@@ -0,0 +1,271 @@
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author threedr3am
+ */
+@RestController
+@RequestMapping(value = "/v2/paramFlow")
+public class ParamFlowRuleControllerV2 {
+
+    private final Logger logger = LoggerFactory.getLogger(ParamFlowRuleControllerV2.class);
+
+    @Autowired
+    private AppManagement appManagement;
+    @Autowired
+    private RuleRepository<ParamFlowRuleEntity, Long> repository;
+
+    @Autowired
+    private AuthService<HttpServletRequest> authService;
+
+    @Autowired
+    @Qualifier("paramFlowRuleDefaultProvider")
+    private DynamicRuleProvider<List<ParamFlowRuleEntity>> ruleProvider;
+    @Autowired
+    @Qualifier("paramFlowRuleDefaultPublisher")
+    private DynamicRulePublisher<List<ParamFlowRuleEntity>> rulePublisher;
+
+    private boolean checkIfSupported(String app, String ip, int port) {
+        try {
+            return Optional.ofNullable(appManagement.getDetailApp(app))
+                .flatMap(e -> e.getMachine(ip, port))
+                .flatMap(m -> VersionUtils.parseVersion(m.getVersion())
+                    .map(v -> v.greaterOrEqual(version020)))
+                .orElse(true);
+            // If error occurred or cannot retrieve machine info, return true.
+        } catch (Exception ex) {
+            return true;
+        }
+    }
+
+    @GetMapping("/rules")
+    public Result<List<ParamFlowRuleEntity>> apiQueryAllRulesForMachine(HttpServletRequest request,
+                                                                        @RequestParam String app,
+                                                                        @RequestParam String ip,
+                                                                        @RequestParam Integer port) {
+        AuthUser authUser = authService.getAuthUser(request);
+        authUser.authTarget(app, PrivilegeType.READ_RULE);
+        if (StringUtil.isEmpty(app)) {
+            return Result.ofFail(-1, "app cannot be null or empty");
+        }
+        if (StringUtil.isEmpty(ip)) {
+            return Result.ofFail(-1, "ip cannot be null or empty");
+        }
+        if (port == null || port <= 0) {
+            return Result.ofFail(-1, "Invalid parameter: port");
+        }
+        if (!checkIfSupported(app, ip, port)) {
+            return unsupportedVersion();
+        }
+        try {
+            return Result.ofSuccess(ruleProvider.getRules(app));
```

Comment:
Here we might need to add logic for handling cluster rules. You could refer to `FlowControllerV2` like this:

```java
 List<FlowRuleEntity> rules = ruleProvider.getRules(app);
            if (rules != null && !rules.isEmpty()) {
                for (FlowRuleEntity entity : rules) {
                    entity.setApp(app);
                    if (entity.getClusterConfig() != null && entity.getClusterConfig().getFlowId() != null) {
                        entity.setId(entity.getClusterConfig().getFlowId());
                    }
                }
            }
            rules = repository.saveAll(rules);
            return Result.ofSuccess(rules);
```